#! /bin/sh /usr/share/dpatch/dpatch-run
## 32_debug_option.dpatch by Petter Reinholdtsen <pere@hungry.com>

Implement new option -D instead of the compile time option DEBUG to
ease debugging.

@DPATCH@
--- insserv-1.11.0.orig/insserv.c
+++ insserv-1.11.0/insserv.c
@@ -107,6 +107,9 @@
 /* When to be verbose */
 static boolean verbose = false;
 
+/* Enable debug output */
+static int debug = 0;
+
 /* When to be verbose */
 static boolean dryrun = false;
 
@@ -1946,6 +1949,7 @@
     {"config",	1, (int*)0, 'c'},
     {"dryrun",	0, (int*)0, 'n'},
     {"default",	0, (int*)0, 'd'},
+    {"debug",	0, (int*)0, 'D'},
     {"remove",	0, (int*)0, 'r'},
     {"force",	0, (int*)0, 'f'},
     {"path",	1, (int*)0, 'p'},
@@ -1993,7 +1997,7 @@
     for (c = 0; c < argc; c++)
 	argr[c] = (char*)0;
 
-    while ((c = getopt_long(argc, argv, "c:dfrhvno:p:", long_options, (int *)0)) != -1) {
+    while ((c = getopt_long(argc, argv, "c:dDfrhvno:p:", long_options, (int *)0)) != -1) {
 	switch (c) {
 	    case 'c':
 		insconf = optarg;
@@ -2002,6 +2006,9 @@
 	    case 'd':
 		defaults = true;
 		break;
+	    case 'D':
+		debug++;
+		break;
 	    case 'r':
 		del = true;
 		break;
@@ -2121,11 +2128,10 @@
 	}
     }
 
-#if defined(DEBUG) && (DEBUG > 0)
-    for (c = 0; c < argc; c++)
-	if (argr[c])
-	    printf("Overwrite argument for %s is %s\n", argv[c], argr[c]);
-#endif /* DEBUG */
+    if (debug > 0)
+        for (c = 0; c < argc; c++)
+	    if (argr[c])
+	        printf("Overwrite argument for %s is %s\n", argv[c], argr[c]);
 
     /*
      * Scan and set our configuration for virtual services.
@@ -2675,10 +2693,10 @@
     if (maxorder > 99)
 	error("Maximum of 99 in ordering reached\n");
 
-#if defined(DEBUG) && (DEBUG > 0)
-    printf("Maxorder %d\n", maxorder);
-    show_all();
-#else
+    if (debug > 0) {
+        printf("Maxorder %d\n", maxorder);
+	show_all();
+    } else {
 # ifdef SUSE	/* SuSE's SystemV link scheme */
     pushd(path);
     for (runlevel = 0; runlevel < RUNLEVLES; runlevel++) {
@@ -2957,7 +2975,7 @@
 	closedir(rcdir);
     }
 # endif /* !SUSE, standard SystemV link scheme */
-#endif  /* !DEBUG */
+    }
 
     /*
      * Do the makedep
--- insserv-1.11.0.orig/listing.c
+++ insserv-1.11.0/listing.c
@@ -454,7 +454,6 @@
 /*
  * For debuging: show all services
  */
-#if defined(DEBUG) && (DEBUG > 0)
 void show_all()
 {
     list_t *tmp;
@@ -468,7 +467,6 @@
 		   dir->start, dir->name, dir->lvl, lvl2str(dir->lvl), *dir->name == '$' ? "system" : "guessed");
     }
 }
-#endif
 
 /*
  * Used within loops to get names not included in this runlevel.
